assets/js/script.js

```diff
@@ -0,0 +1,120 @@
+//******************************DOM OBJECTS******************************
+const newTaskInput = document.getElementById("new-task-input");
+const addTaskBtn = document.getElementById("addBtn");
+const searchBar = document.getElementById("search");
+const table = document.getElementById("table-body");
+const counter = document.getElementById("tasks-counter");
+
+var tasksArray = []; //this array contains all the todo list tasks
+
+//****************************EVENT LISTENERS****************************
+/* on the first window load, fetch data from API and save it in sessionStorage
+   on further loads, fetch the data from sessionStorage */
+window.onload = () => {
+  if (!sessionStorage.getItem("visited")) {
+    sessionStorage.setItem("visited", true); // Set visited flag in sessionStorage
+    fetchData();
+  } else {
+    let savedData = sessionStorage.getItem("savedTasks");
+    if (savedData) {
+      tasksArray = JSON.parse(savedData);
+      displayTasks(tasksArray);
+    }
+  }
+};
+
+//search for a task
+searchBar.onkeyup = () => {
+  let searchQuery = searchBar.value.toLowerCase(); //get the search query
+  let filteredTasksArray = tasksArray.filter((task) =>
+    task.todo.toLowerCase().includes(searchQuery)
+  );
+  displayTasks(filteredTasksArray);
+};
+
+//add a new task
+addTaskBtn.onclick = () => {
```

Comment:
```
// Add a new task
function addTask() {
  const newTask = newTaskInput.value.trim();
  if (!newTask) {
    alert("Please enter the task");
    return;
  }
  const task = {
    id: tasksArray.length + 1,
    todo: newTask,
    userId: "00",
    completed: false,
  };
  tasksArray.push(task);
  saveTasksToSessionStorage();
  displayTasks(tasksArray);
  newTaskInput.value = "";
}

```
Avoid Inline Event Handlers:
Instead of using inline event handlers like onclick, it's generally better to use addEventListener. This separates the behavior from the HTML structure and allows for better

Reply:
also apply this for all other events